_posts/2019/2019-02-24-secure-your-staging-environment-with-basic-auth.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Secure Your Rails Staging Environment with HTTP Basic Authentication
+description: Opening up your development environment is a bad idea
+layout: article
+category: ruby
+image:
+  base: '2019/secure-your-staging-environment-with-basic-auth'
+  alt: 'Yellow padlock'
+  source: "https://unsplash.com/photos/0Yiy0XajJHQ"
+  credit: "Chris Panas"
+---
+
+A sensible approach to testing features before you ship them to customers is to use a staging environment that closely resembles, your live, customer-facing, production application.
+
+It’s important to make sure that only people inside your organisation can use this testing environment, it might contain broken features (hopefully not!) or
+
+
+## Instead of…
+
+…leaving your staging environment open for anyone who finds it on the wild internet.
+
+
+## Use…
+
+…the basic HTTP authentication in Rails to secure the whole application from non-authorised users.
+
+```ruby
+class ApplicationController < ActionController::Base
+  http_basic_authenticate_with name: ENV["HTTP_BASIC_USER"],
+                               password: ENV["HTTP_BASIC_PASSWORD"],
+                               if: -> { ENV["HTTP_BASIC_PASSWORD"].present? }
+end
+```
+
+…and set the environment (`ENV`) variables for `HTTP_BASIC_USER` and `HTTP_BASIC_PASSWORD` in your staging environment.
+
+
+## But why?
+
+Giving people access to your staging environment is a bad idea. It might contain unreleased or bug-ridden features. Or perhaps even allow access to the functionality of you application for free as staging environments are often linked up to sandboxed versions of payment processors for testing.
```

Comment:
*Furthermore, it might even allow free access to normally paid-for functionality in your application, since staging environments are often linked up to...